packages/cli/src/commands/dev/dev.ts

```diff
@@ -27,76 +25,59 @@ export default async function dev(
   let cwd = resolve(dir);
   const listen = parseListen(opts['--listen'] || '3000');
 
-  // retrieve dev command
-  let [link, frameworks] = await Promise.all([
-    getLinkedProject(client, cwd),
-    getFrameworks(client),
-  ]);
-
-  if (link.status === 'not_linked' && !process.env.__VERCEL_SKIP_DEV_CMD) {
-    link = await setupAndLink(client, cwd, {
-      autoConfirm: opts['--confirm'],
-      successEmoji: 'link',
-      setupMsg: 'Set up and develop',
-    });
-
-    if (link.status === 'not_linked') {
-      // User aborted project linking questions
+  let project = await readProjectSettings(join(cwd, VERCEL_DIR));
+  // If there are no project settings, only then do we pull them down
+  while (!project?.settings && !process.env.__VERCEL_SKIP_DEV_CMD) {
+    const confirmed = await confirm(
+      `No Project Settings found locally. Run ${getCommandName(
+        'pull'
+      )} for retrieving them?`,
+      true
+    );
+    if (!confirmed) {
+      client.output.print(`Aborted. No Project Settings retrieved.\n`);
       return 0;
     }
+    const result = await pull(client);
+    if (result !== 0) {
+      return result;
+    }
+    project = await readProjectSettings(join(cwd, VERCEL_DIR));
   }
 
-  if (link.status === 'error') {
-    return link.exitCode;
+  cwd = project?.settings.rootDirectory
+    ? join(cwd, project.settings.rootDirectory)
+    : cwd;
+
+  process.chdir(cwd);
+
+  const framework = findFramework(project?.settings.framework);
+  // If this is undefined, we bail. If it is null, then findFramework should return "Other",
+  // so this should really never happen, but just in case....
+  if (framework === undefined) {
```

Comment:
What about when `__VERCEL_SKIP_DEV_CMD` is assigned?

Reply:
Good point. Will add condition